Android/app/src/main/java/com/example/instagramclone/Adapters/ViewHolders/PostsAdapter.java

```diff
@@ -0,0 +1,72 @@
+package com.example.instagramclone.Adapters.ViewHolders;
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.instagramclone.Activities.MainActivity;
+import com.example.instagramclone.Common.Data;
+import com.example.instagramclone.Fragments.OnePostFragment;
+import com.example.instagramclone.Models.FeedItemModel;
+import com.example.instagramclone.Models.UserModel;
+import com.example.instagramclone.R;
+import com.squareup.picasso.Picasso;
+
+import java.util.ArrayList;
+
+public class PostsAdapter extends RecyclerView.Adapter<PostsAdapter.PostsViewHolder> {
+
+    private ArrayList<FeedItemModel> postList;
+    private Context context;
+
+    public PostsAdapter(ArrayList<FeedItemModel> postList) {
+        this.postList = postList;
+        this.context = context;
+    }
+
+    @NonNull
+    @Override
+    public PostsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_user_profile_post, parent, false);
+
+        return new PostsViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull PostsViewHolder holder, int position) {
+        Bitmap bitmap = BitmapFactory.decodeByteArray(postList.get(position).getImageBuffer().getData(), 0, postList.get(position).getImageBuffer().getData().length);
```

Comment:
Why don't you use Glide or Picasso for loading images? Also server should return you an image url after upload instead of storing in a buffer

Reply:
The server is hosted on Heroku, in the free tier, so the dyno gets wiped with every deploy, deleting all files and photos which would have been stored there - making it impossible to just save the photos on the disk, and return their urls. Storing the photos as buffers in the database seemed as a better option, since it's permanent

Reply:
Then @adammaran could have used firebase cloud storage for that, or even implement it on backend side but not sure if it would work

Reply:
Well yes, but I'm not sure if it was allowed to use firebase, as we were supposed to create and model our own databases for the project

Reply:
As professor would know 😂

Reply:
We talked about uploading the images to cloud firestore and storing the url in the database object but didnt excecute the idea